src/components/connections/EditConnectionView.ts

```diff
@@ -0,0 +1,53 @@
+import { ExtensionContext, window, ViewColumn, Uri } from "vscode";
+import mustache = require("mustache");
+import * as path from "path";
+import { View } from "../View";
+import { InfluxDBTreeDataProvider, InfluxDBConnection } from "./Connection";
+
+export class EditConnectionView extends View {
+  public constructor(context: ExtensionContext) {
+    super(context, "templates/editConn.mst");
+  }
+
+  public async show(
```

Comment:
I would extract this into showEdit and showNew instead of passing a boolean, then the common logic can go into a private method